### PR TITLE
add initial guide of the Conversion Webhook check

### PR DESCRIPTION
**Pull request by adenitiu**
**What this PR does / why we need it**:
Adds docs on how the end-users can fix their CRDs with Conversion Webhook.

**Which issue(s) this PR fixes**:
Added in order to handle landscape-issue https://github.tools.sap/kubernetes/landscape-issues/issues/97

**Special notes for your reviewer**:

**Release note**:

```noteworthy user
Fix problematic CRDs with Conversion Webhooks
```

### CODE REVIEW

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+This creates a mismatch between the requested and stored version for all clients (kubectl, KCM, etc.). When the CRD also declares the usage of a conversion webhook, it gets called whenever a client requests information about a resource that still exists in the old version. If the CRD is created by the end-user, the webhook runs on the shoot side, whereas controllers / kapi-servers run separated, as part of the control-plane. For the webhook to be reachable a working VPN connection seed -> shoot is essential. In scenarios where the VPN connection is broken, **kube-controller-manager** eventually stops its garbage collection as that requires to list `v1.PartialObjectMetadata` for everything to build a dependency graph. Without kube-controller-manager Garbage Collector, managed resources get stuck during update/rollout.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:14]:
```suggestion
This creates a mismatch between the requested and stored version for all clients (kubectl, KCM, etc.). When the CRD also declares the usage of a conversion webhook, it gets called whenever a client requests information about a resource that still exists in the old version. If the CRD is created by the end-user, the webhook runs on the shoot side, whereas controllers / kapi-servers run separated, as part of the control-plane. For the webhook to be reachable, a working VPN connection `seed -> shoot` is essential. In scenarios where the VPN connection is broken, the **kube-controller-manager** eventually stops its garbage collection, as that requires it to list `v1.PartialObjectMetadata` for everything to build a dependency graph. Without the kube-controller-manager's garbage collector, managed resources get stuck during update/rollout.
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+## Breaking situations
+
+When a user upgrades to `failureTolerance: node|zone`, that will cause the VPN **deployments** to be _replaced_ by **statefulsets**. However, as the VPN connection is broken upon teardown of the deployment, garbage collection will fail leading to a situation that is stuck until an operator manually tackles it.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:18]:
```suggestion
When a user upgrades to `failureTolerance: node|zone`, that will cause the VPN **deployments** to be _replaced_ by **statefulsets**. However, as the VPN connection is broken upon teardown of the deployment, garbage collection will fail, leading to a situation that is stuck until an operator manually tackles it.
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+Such a situation can be avoided, if the end-user has correctly configured CRDs containing conversion webhooks.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:20]:
```suggestion
Such a situation can be avoided if the end-user has correctly configured CRDs containing conversion webhooks.
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+## Check problematic CRDs
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:22]:
```suggestion
## Checking Problematic CRDs
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+done
+echo "Problematic CRDs are reported above."
+```
+
+
+# Resolve CRDs
+
+Below we give the steps needed to be taken in order to fix the CRDs reported by the script above.
+
+Inspect all your CRDs that have conversion webhooks in place. If you have more than 1 version defined in its `spec.status.storedVersions` field, then initiate migration as described in the K8S guide https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version , option2.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:73]:
```suggestion
Inspect all your CRDs that have conversion webhooks in place. If you have more than 1 version defined in its `spec.status.storedVersions` field, then initiate migration as described in **Option 2** in the [Upgrade existing objects to a new stored version](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version) guide.
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+For convenience, we provided the necessary steps below.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:75]:
```suggestion
For convenience, we have provided the necessary steps below.
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+{{% alert color="info" title="Note" %}}
+Please test the following steps on a non-productive landscape, to make sure that the new CR version doesn’t break any of your existing workload.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:78]:
```suggestion
Please test the following steps on a non-productive landscape to make sure that the new CR version doesn’t break any of your existing workloads.
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+{{% /alert %}}
+
+
+1. Please check/set the old CR version has `storage:false` and set the new CR version to `storage:true`.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:82]:
```suggestion
1. Please check/set the old CR version to `storage:false` and set the new CR version to `storage:true`.
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+    For sake of example, let’s consider 2 versions `v1beta1` (old) and `v1` (new).
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:84]:
```suggestion
    For the sake of an example, let’s consider the two versions `v1beta1` (old) and `v1` (new).
```

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    - name: v1beta1
+    ......
+    storage: false
+
+    - name: v1
+    ......
+    storage: true
+    ```
+
+2. Convert custom-resources to newest version.
```
Comment (n-boshnakov) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md:112]:
```suggestion
2. Convert `custom-resources` to the newest version.
```

**Review by n-boshnakov — CHANGES_REQUESTED** [review]
Thank you for your contribution, Andreea! I have made some stylistic suggestions for your PR in order to keep it more in-line with the already existing documentation.

File: website/documentation/guides/administer-shoots/conversion-webhook/_index.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
+---
+title: Fix Problematic Conversion Webhooks
+level: advanced
+category: Operation
+scope: users
+---
+
+## Reasoning
+
+**Custom Resource Definition (CRD)** is what you use to define a `Custom Resources`. This is a powerful way to extend Kubernetes capabilities beyond the default installation, adding any kind of API object useful for your application.
```
Comment (dguendisch) [inline website/documentation/guides/administer-shoots/conversion-webhook/_index.md]:
```suggestion
**Custom Resource Definition (CRD)** is what you use to define a `Custom Resource`. This is a powerful way to extend Kubernetes capabilities beyond the default installation, adding any kind of API object useful for your application.
```

**Review by dguendisch — CHANGES_REQUESTED** [review]

**Review by dguendisch — APPROVED** [review]
/lgtm

**Pull request merged**